src/core/ddsc/src/dds_qos.c

```diff
@@ -335,6 +335,152 @@ void dds_qset_ignorelocal (dds_qos_t * __restrict qos, dds_ignorelocal_kind_t ig
   qos->present |= QP_CYCLONE_IGNORELOCAL;
 }
 
+void dds_qset_prop (dds_qos_t * __restrict qos, const char * __restrict name, const char * __restrict value)
+{
+  bool found = false;
+  if (qos == NULL || name == NULL || value == NULL)
+    return;
+
+  if (!(qos->present & QP_PROPERTY_LIST) || qos->property.value.n == 0)
+  {
+    qos->property.value.n = 0;
+    qos->property.value.props = dds_alloc (sizeof (*qos->property.value.props));
+    if (!(qos->present & QP_PROPERTY_LIST))
+    {
+      qos->property.binary_value.n = 0;
+      qos->property.binary_value.props = NULL;
+    }
+  }
+  else
+  {
+    for (uint32_t i = 0; !found && i < qos->property.value.n; i++)
+    {
+      if (strcmp (qos->property.value.props[i].name, name) == 0)
+      {
+        dds_free (qos->property.value.props[i].value);
+        qos->property.value.props[i].value = dds_string_dup (value);
+        found = true;
+      }
+    }
+    if (!found)
+      qos->property.value.props = dds_realloc (qos->property.value.props,
+        (qos->property.value.n + 1) * sizeof (*qos->property.value.props));
+  }
+  if (!found)
+  {
+    qos->property.value.props[qos->property.value.n].propagate = 0;
+    qos->property.value.props[qos->property.value.n].name = dds_string_dup (name);
+    qos->property.value.props[qos->property.value.n].value = dds_string_dup (value);
+    qos->property.value.n++;
+    qos->present |= QP_PROPERTY_LIST;
+  }
+}
+
+void dds_qunset_prop (dds_qos_t * __restrict qos, const char * __restrict name)
+{
+  if (qos == NULL || !(qos->present & QP_PROPERTY_LIST) || qos->property.value.n == 0 || name == NULL)
+    return;
+
+  for (uint32_t i = 0; i < qos->property.value.n; i++)
```

Comment:
I would prefer using a lookup function that returns an index (or "not found" in some way) — and then use that in the setters as well.

Reply:
You're right, this simplifies the get/set/unset, thanks!